### DISCUSSION TITLE
Code coverage of the project

### DISCUSSION THREAD

**Discussion by nguoithichkhampha** [nlohmann/json#4054] [Show and tell]
I tried to check code coverage if this repo. 
line coverage is almost same but branch coverage is different and function coverage is quite low.
Can someone help me to explain 
![Screenshot from 2023-06-15 11-00-25](https://github.com/nlohmann/json/assets/1573602/5d552903-d942-4af4-a954-e8780f9598f2)

I build with this config
`cmake -DCMAKE_BUILD_TYPE=Debug -DJSON_BuildTests=ON -DCMAKE_CXX_FLAGS="--coverage" ../`
and generate the report by gcovr
`gcovr -r ../include/ . --exclude-noncode-lines -j5 --print-summary --html-details coverage/index.html --exclude-unreachable --exclude-throw-branches`

**Comment by nlohmann** [comment]
Can you share the HTML report? It would be interesting to see which functions are not called.

**Comment by nguoithichkhampha** [comment]
[coverage.zip](https://github.com/nlohmann/json/files/11753508/coverage.zip)
Here is the report. another thing, do you know why function not called but line coverage still count ?

**Reply by nlohmann** [reply]
I guess this is all related to templates.
